hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/DataFrameUtil.scala

```diff
@@ -31,7 +33,7 @@ object DataFrameUtil {
    */
   def createFromInternalRows(sparkSession: SparkSession, schema:
   StructType, rdd: RDD[InternalRow]): DataFrame = {
-    val logicalPlan = LogicalRDD(schema.toAttributes, rdd)(sparkSession)
+    val logicalPlan = LogicalRDD(SparkAdapterSupport.sparkAdapter.toAttributes(schema), rdd)(sparkSession)
```

Comment:
StructType.toAttributes was removed in Spark 3.5.0 by [SPARK-44353](https://github.com/apache/spark/pull/41925)

Solution is to switch to use DataTypeUtils.toAttributes